avocado/core/job.py

```diff
@@ -204,6 +204,44 @@ def _setup_job_results(self):
             id_file_obj.flush()
             os.fsync(id_file_obj)
 
+        self._setup_job_category()
+
+    def _setup_job_category(self):
+        """
+        This has to be called after self.logdir has been defined
+
+        It attempts to create a directory one level up from the job results,
+        with the given category name.  Then, a symbolic link is created to
+        this job results directory.
+
+        This should allow a user to look at a single directory for all
+        jobs of a given category.
+        """
+        category = getattr(self.args, 'job_category', None)
+        if category is None:
+            return
+
+        if category != astring.string_to_safe_path(category):
+            LOG_JOB.warning("Unable to set category in job results: name is not "
+                            "filesystem safe: %s", category)
```

Comment:
Where would this warning present itself?

Rather than an _assert_ here, would it be better to just use the output of `astring.string_to_safe_path(category)`?

Reply:
Indeed, that was something I initially considered.  But I feel that behavior would (and can still) be driven by a different switch.

Maybe we can evolve this with options such as `--job-category-use-safe-path`.  Other related options include ``--job-category-fail-job` if we fail to create the category.

Anyway, I guess those can be added on top of this.

Reply:
Missed the question:

> Where would this warning present itself?

Because our developer here testing this tried to use a dir/file path and could not find that warning anywhere and rather avocado seemed to just exit silently without creating the category dir.

Reply:
True... and I guess there's something going on with the "avocado.test" logger... I'm investigating it now.